[MRG]: tight bounds in contours

GaelVaroquaux opened the pull request:
A test implementation for tight bounds in contours.

Attempts to fix #1450

GaelVaroquaux commented:
The latest commit is a bit heuristic.

These changes will need to be war-tested on a bunch of different usage situations.

KamalakerDadi commented:
I had a quick look at the plot_prob_atlas example. There is some problem with displaying contours.For instance, a msdl atlas in https://2495-1235740-gh.circle-artifacts.com/0/home/ubuntu/nilearn/doc/_build/html/auto_examples/01_plotting/plot_prob_atlas.html#sphx-glr-auto-examples-01-plotting-plot-prob-atlas-py

In plot_prob_atlas, we add an heuristic to get the threshold (levels) for displaying. May be we are double thresholding. I might also be wrong. I can have a closer look in more detail, later.

GaelVaroquaux commented:
I changed the strategy to something that should be more robust.

I believe that this is now ready for merge.

@fliem, do you mind testing this again, as I changed the strategy. It should work fine.

KamalakerDadi commented on doc/whats_new.rst:
```diff
@@ -33,6 +33,8 @@ Enhancements
 
     - Reworking and renaming of connectivity measures example. 
 
+    - Tighter bounding boxes when using add_contour for plotting.
```
add_contour -> add_contours. Just to be consistence with naming of our method.

KamalakerDadi commented on nilearn/plotting/displays.py:
```diff
@@ -658,11 +658,22 @@ def _map_show(self, img, type='imshow',
             xmin_, xmax_, ymin_, ymax_, zmin_, zmax_ = \
                 get_mask_bounds(new_img_like(img, not_mask, affine))
 
+
```
Just to avoid PEP8. Can you please remove this ?

KamalakerDadi commented:
Do you have any tests to add ?

GaelVaroquaux commented:
> Do you have any tests to add ?

I couldn't think of a way to add a test here.

bthirion commented:
LGTM.

fliem commented:
Output still looks good.

GaelVaroquaux commented:
> Output still looks good.

Great. Once I get @KamalakerDadi's +1, we'll merge.

KamalakerDadi commented:
+1 from my side. Merging this one.
